runtime/metricsview/executor_timestamps.go

```diff
@@ -0,0 +1,243 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	defaultExecutionTimeout = time.Minute * 3
+)
+
+func (e *Executor) resolveDuckDB(ctx context.Context) (time.Time, time.Time, time.Time, error) {
+	timeDim := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+
+	var watermarkExpr string
+	if e.metricsView.WatermarkExpression != "" {
+		watermarkExpr = e.metricsView.WatermarkExpression
+	} else {
+		watermarkExpr = fmt.Sprintf("MAX(%s)", timeDim)
+	}
+
+	rangeSQL := fmt.Sprintf(
+		"SELECT min(%[1]s) as \"min\", max(%[1]s) as \"max\", %[2]s as \"watermark\", max(%[1]s) - min(%[1]s) as \"interval\" FROM %[3]s %[4]s",
+		timeDim,
+		watermarkExpr,
+		escapedTableName,
+		e.security.RowFilter(),
```

Comment:
This is not prefixed with a `WHERE` clause, so would break if there's a row filter.

We have a new test framework for resolvers that supports easily testing different OLAPs and different user attributes / security policies. See here:
- For more details: runtime/resolvers/testdata/README.md
- For an example: runtime/resolvers/testdata/metrics_comparisons.yaml